backend_jobs/ingestion_pipeline/unit_tests/image_provider_interface_test.py

```diff
@@ -0,0 +1,63 @@
+"""
+  Copyright 2020 Google LLC
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+"""
+
+import unittest
+from datetime import datetime
+from backend_jobs.ingestion_pipeline.pipeline_lib import data_types
+from backend_jobs.ingestion_pipeline.providers.image_provider_flickr import FlickrProvider
+
+IMAGE_PROVIDERS = {'FlickrProvider': FlickrProvider()}
+
+
+class TestProviderIntrface(unittest.TestCase):
+    """ This class is in charge of testing the image provider interface.
+    """
+
+    def test_get_images_and_exract_attributes(self):
+        """ This test checks both the get_images and the get_image_attributes functions'
+        implementation for all providers.
+        We test, given an image when calling the get_image_attributes function,
+        that the output is in the right format.
+        """
+        for provider in IMAGE_PROVIDERS:
+            current_provider = IMAGE_PROVIDERS[provider]
+            images = current_provider.get_images(1)
+            for image in images:
+                image_attributes = current_provider.get_image_attributes(image)
+                self.assertIsInstance(image_attributes, data_types.ImageAttributes)
+                self.assertIsInstance(image_attributes.image_id, (str, type(None)))
```

Comment:
can we really have missing ID's here?

Reply:
we have in the start id-None,
it is changed only in the generate image id stage